libevmasm/Assembly.cpp

```diff
@@ -1465,6 +1476,18 @@ LinkerObject const& Assembly::assembleEOF() const
 		setBigEndianUint16(ret.bytecode, codeSectionSizePositions[codeSectionIndex], ret.bytecode.size() - sectionStart);
 	}
 
+	for (auto const& [refPos, tagId]: tagRef)
+	{
+		solAssert(tagId < m_tagPositionsInBytecode.size(), "Reference to non-existing tag.");
+		size_t tagPos = m_tagPositionsInBytecode[tagId];
+		solAssert(tagPos != std::numeric_limits<size_t>::max(), "Reference to tag without position.");
+
+		ptrdiff_t const relativeJumpOffset = static_cast<ptrdiff_t>(tagPos) - (static_cast<ptrdiff_t>(refPos) + 2);
+		solRequire(-0x8000 <= relativeJumpOffset && relativeJumpOffset <= 0x7FFF, AssemblyException, "Relative jump too far");
+		solRequire(0 <= relativeJumpOffset || relativeJumpOffset < -2, AssemblyException, "Relative jump offset into immediate argument.");
```

Comment:
Just a small nitpick: can we keep the conditions ordered like intervals (`(..., -2)`, `[0, ...)`)? I think it's much more natural to read:

```suggestion
		solAssert(relativeJumpOffset < -2 || 0 <= relativeJumpOffset, AssemblyException, "Relative jump offset into immediate argument.");
```

Also, unlike the check above, this one is an assert - we're not supposed to ever generate code that fails it, regardless of what the user input is.

Reply:
OK.